### DISCUSSION TITLE
fabric8 create op does not support providing field manager?

### DISCUSSION THREAD

**Discussion by fhalde** [fabric8io/kubernetes-client#5984] [General]
Is there a reason why we can't supply a field manager ( or override the default `fabric8-kubernetes-client` ) when trying to "create" a resource through fabric8?

kubectl does seem to support it
```
kubectl create -h
   --field-manager='kubectl-create':
	Name of the manager used to track field ownership.
```

The only operation after providing a `fieldManager(String)` is `serverSideApply()` which is a bit problematic for my controller code path that strictly expects the "create" to fail if a resource already exists so that appropriate error can be shown to the user ( regarding resource name clash ).

**Comment by fhalde** [comment]
unless ofcourse i've missed something :)

**Comment by fhalde** [comment]
it's important to point out that our JVM process is comprised of multiple controllers that share a single kubernetes client ( we use josdk ). so a global kubernetes-client property won't do for us

**Comment by shawkins** [comment]
> The only operation after providing a fieldManager(String) is serverSideApply() which is a bit problematic for my controller code path that strictly expects the "create" to fail if a resource already exists so that appropriate error can be shown to the user ( regarding resource name clash ).

If you set a dummy resourceVersion, such as "1", then it will only allow the serverSideApply to succeed if the resource does not already exist. This is the mechanism that is used by the java operator sdk.

**Comment by fhalde** [comment]
thanks @shawkins , that would do for now :)
